### PR TITLE
Add seasonality measure and action

### LINKED ISSUE #55: Explore seasonality trends by extending to earlier dates
Upon visualising the report, we have noticed an increase in counts across all conditions during the summer months (May-Aug). 
The hypothesis is that these trends may be due to the seasonality of certain conditions, and thus to explore this idea, we have extended the study period of the report to include the latest month (October) to see whether trends in seasonal conditions.

In order to extend to earlier dates prior to PF launch, we will have to visualise clinical events that are **not** linked to PF. This is because pre launch dates will have minimal counts if linked to PF.

Conditions:

1. Infected Insect Bites
 -  More common during spring and summer months

2. Sinusitis 
 - More common during winter months

3. Sore throat
 - More frequent during winter months

4. Otitis media
 - Higher incidence during autumn and winter

5. Impetigo
 - Usually more prevalent in summer

6. Urinary tract infection
 - Most studies indicate summer months peak, a few also mention winter seasonality?

7. Shingles
 - Not considered seasonal.

### LINKED ISSUE #58: Fix selection of `condition_events`
We should use `selected_events` (and not `clinical_events`) in all `where()` statements here:

https://github.com/opensafely/pharmacy-first/blob/0e05a983581f0b4678995b17374ae1808b71a466/analysis/measures_definition_pf_codes_conditions.py#L130-L131

### PR DESCRIPTION
**Pull request by viv3ckj**
Closes #55 
Closes #58

### CODE REVIEW

**Comment by viv3ckj** [comment]
I decided to add a new seasonality measures file instead of adding it into the existing measures file, for the following reasons:

- Because we are addressing seasonality as opposed to PF usage, I thought it best to keep different logic separate, as this will improve readability.

- As this project is a work in progress, we may need to add to the PF measures file, which may complicate things, especially with regards to memory issues while running jobs. Modularising code helps this, and will allow us to only run code that we need. 

- Maintaining code that is split into different files in this case might make things more efficient, as we can isolate and fix issues easily without breaking or complicating the other measures. 

- We already have ideas on how to add to the seasonality analysis to look for additional details. Modularising code like this improves scalability, as we can add logic to the file without making the overall file unnecessarily complex and hard to debug.

File: analysis/measures_definition_pf_seasonality.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Select all pharmacy first events from selected events
+pharmacy_first_events = selected_events.where(
+    selected_events.snomedct_code.is_in(
+        pharmacy_first_event_codes["combined_pf_service"]
+    )
+)
+
+# Create variable which contains boolean values of whether pharmacy first event exists for patient
+has_pharmacy_first = pharmacy_first_events.exists_for_patient()
```
Comment (milanwiedemann) [inline analysis/measures_definition_pf_seasonality.py:31]: I only wondered if we should shorten this or do we want to have `pharmacy_first_events` as a separate object?
```suggestion
# Create variable which contains boolean values of whether pharmacy first event exists for patient
has_pharmacy_first = selected_events.where(
    selected_events.snomedct_code.is_in(
        pharmacy_first_event_codes["combined_pf_service"]
    )
).exists_for_patient()
```

**Review by milanwiedemann — APPROVED** [review]
Thanks, this looks great! I just wondered if you want to shorten the way we get to `has_pharmacy_first`? Up to you, I have no preference!

File: analysis/measures_definition_pf_seasonality.py (hunk 1 above)
Comment (viv3ckj) [reply analysis/measures_definition_pf_seasonality.py]: I'll make this change!

**Pull request merged**